### ISSUE TITLE
manage yearly code updates

### ISSUE THREAD

**Issue by stephenwaite** [openemr/openemr#2280]
from the forum:
https://community.open-emr.org/t/icd-10-2019-unsupported-database-load-file/11554/21

thinking that a patch might be able to handle the md5sum inserts into the table and would then just need to allow user to move files in and out of the contrib directory with a gui?

**Comment by danehrlich1** [comment]
Jerry:

Can you write out in a little more detail what your idea would do? Maybe a few lines of pseudo code?

I was working with Scott Arciszewski to do auto updates over the internet for OpenEMR source code (quite tricky) but I think this is a totally separate issue...if not we could also use the interface being designed for that for this?

**Comment by sjpadgett** [comment]
md5sum and file integrity is really a user responsibility during/after download so that requirement could go away imo. The reason this interface is the way it is, is because there is licensing involved especially for rxnorm and snomeds(U.S free but not International). 

@danehrlich1 "Maybe a few lines of pseudo code?"
hmm, I've train many budding engineers over the years and one thing that will do them a disservice is holding their hand. Assign the task with some guidelines and let them do the engineering. Otherwise they become coders. Nothing wrong with being a coder but I bet most these guys/gals have their sights set higher. Okay my friend i'm done imparting unwanted wisdom. :) I will set out some guidelines later today once I review that code again. Peace.....

**Comment by bradymiller** [comment]
hi @stephenwaite , @danehrlich1 , an @sjpadgett ,

When we wrote this in the past, in order to make things simple, we required that only the version that was getting imported could be in the directory (so users need to add the new and delete the old), but we don't need to do this.

The thing that likely needs to be fixed in codebase  is to allow more than one version of icd10 to sit in the directory and to only "use" the most recent version that is there (and ignore the others). If we fixed that, then supporting new versions of ICD10 codesets in the patch would be straightforward.

Note this is specific for ICD10 since licensing allows us to do that. The other ones, snomed, rxnorm, have licensing that does not allow us to include them in our codebase or distribute them.
